pycytominer_transform/convert.py

```diff
@@ -2,349 +2,254 @@
+        #   updated_column_name: 'Cells_AreaShape_Area'
+        if column_name not in identifying_columns and not column_name.startswith(
+            source_group_name_stem
+        ):
+            updated_column_names.append(f"{source_group_name_stem}_{column_name}")
+
+        # if-condition for prepending 'Metadata' and source_group_name_stem to column name
+        # where colname is an identifying column
+        # and where the colname does not already start with 'Metadata_'
+        # and colname not in metadata list
+        # and colname does not include 'ObjectNumber'
+        # for example:
+        #   source_group_name_stem: 'Cells'
+        #   column_name: 'Parent_Nuclei'
+        #   updated_column_name: 'Metadata_Cells_Parent_Nuclei'
+        elif (
+            column_name in identifying_columns
+            and not column_name.startswith("Metadata_")
+            and not any(item.capitalize() in column_name for item in metadata)
+            and not "ObjectNumber" in column_name
+        ):
+
+            updated_column_names.append(
+                f"Metadata_{source_group_name_stem}_{column_name}"
+            )
+        # if-condition for prepending 'Metadata' to column name
+        # where colname doesn't already start with 'Metadata_'
+        # and colname is in identifying columns list
+        # for example:
+        #   column_name: 'ObjectNumber'
+        #   updated_column_name: 'Metadata_ObjectNumber'
+        elif (
+            not column_name.startswith("Metadata_")
+            and column_name in identifying_columns
+        ):
+            updated_column_names.append(f"Metadata_{column_name}")
+
+        # else we add the existing colname to the updated list as-is
+        else:
+            updated_column_names.append(column_name)
+
+    # perform table column name updates
+    source["table"] = source["table"].rename_columns(updated_column_names)
 
-    return record
+    return source
 
 
 @task
-def concat_record_group(
-    record_group: List[Dict[str, Any]],
+def concat_source_group(
+    source_group: List[Dict[str, Any]],
     dest_path: str = ".",
-    infer_common_schema: bool = True,
+    common_schema: Optional[List[Tuple[str, str]]] = None,
 ) -> List[Dict[str, Any]]:
     """
-    Concatenate group of records together as unified dataset.
+    Concatenate group of sources together as unified dataset.
+
+    For a reference to data concatenation within Arrow see the following:
+    https://arrow.apache.org/docs/python/generated/pyarrow.concat_tables.html
 
     Args:
-      record_group: List[Dict[str, Any]]:
-        Data structure containing grouped data for concatenation.
-      dest_path: Optional[str] (Default value = None)
-        Optional destination path for concatenated records.
-      infer_common_schema: bool (Default value = True)
-        Whether to infer a common schema when concatenating records.
+        source_group: List[Dict[str, Any]]:
```

Comment:
Your code appears to expect a very specific folder structure; the lines below where you reference `.parent.parent` indicate to me that the method could fail if they don't follow that structure. Perhaps you could describe it in the documentation with an example. If `source_group` is the output of some other method in your code, it'd be good to call out which method you expect to produce the value that's then passed in here.

Reply:
Thank you, great point, I'll add an example to the docs here. I feel the latter half of this comment maybe would be best addressed within #26 , which I'll be sure to implement towards closure there. I feel it's also worth calling out that this structure may be subject to change depending on performance implications (benchmarking is likely needed).

Reply:
Docs addressed within [8227461](https://github.com/cytomining/pycytominer-transform/pull/22/commits/8227461dadc12a310a1fd7005d12635c800b9f12). Other aspects to be covered with future work.